core/src/commons/Data.cpp

```diff
@@ -217,8 +222,24 @@ size_t Data::get_num_rows() const {
   return num_rows;
 }
 
+size_t Data::get_num_outcomes() const {
+  if (outcome_index.has_value()) {
+    return outcome_index.value().size();
+  } else {
+    return 1;
+  }
+}
+
 double Data::get_outcome(size_t row) const {
-  return get(row, outcome_index.value());
+  return get(row, outcome_index.value()[0]);
+}
+
+Eigen::VectorXd Data::get_outcomes(size_t row) const {
+  Eigen::VectorXd out(outcome_index.value().size());
+  for (size_t i = 0; i < outcome_index.value().size(); i++) {
+    out(i) = get(row, outcome_index.value()[i]);
```

Comment:
Given calls to the virtual method `get` can be expensive (as you found when profiling), should we consider introducing a multi-column `get` ?

Reply:
Do you mean a `get_columns` which would give a slice from column i to j?

Should we consider it in #747 since it will also involve a bigger change to the Data class?

Reply:
Yes, and I think it could be a pretty self-contained change? I'm happy to consider it later though to keep this PR focused.

Reply:
How would you do it without having `get_columns` be virtual?

Reply:
I was thinking that `get_columns` would be still virtual, but by passing all indices are once we would avoid repeated calls to `get`. But now I remember that the virtual method lookup isn't what's slow, the problem is that it prevents compiler optimizations like inlining (https://softwareengineering.stackexchange.com/questions/191637/in-c-why-and-how-are-virtual-functions-slower). So my suggestion won't help, I agree we should keep it as-is and focus on #747 in a follow-up.

Reply:
Ok, thank you! That sounds good

Note Travis may fail when this is merged to master because in that deployment "pkgdown" is installed to build the online docs, and that package currently fails installing (one of its myriad of dependencies has a problem).